open_discussions/permissions.py

```diff
@@ -1,14 +1,32 @@
 """Custom permissions"""
 from prawcore.exceptions import (
     Forbidden as PrawForbidden,
-    Redirect as PrawRedirect,
+    Redirect as PrawRedirect
 )
 from rest_framework import permissions
+from rest_framework.generics import get_object_or_404
 
 from channels.models import Channel
 from open_discussions import features
 
 
+def channel_exists(view):
+    """
+    Return True if a Channel object exists for a channel_name in the view, or there is no channel name.
+    Raises 404 if the Channel does not exist.
+
+    Args:
+        view (rest_framework.views.APIView): django DRF view
+
+    Returns:
+        bool: True if Channel exists (or there is no channel name)
+    """
+    channel_name = view.kwargs.get('channel_name', None)
+    if channel_name:
+        get_object_or_404(Channel, name=channel_name)
```

Comment:
Can you change this to an `exists()`, so the overhead is a little lower since we'll be executing this for everything?

Reply:
Change has been made